src/main/scala/dsptools/numbers/chisel_types/UIntTypeClass.scala

```diff
@@ -26,11 +26,11 @@ trait UIntRing extends Any with Ring[UInt] with hasContext {
   }
   override def minus(f: UInt, g: UInt): UInt = {
     val diff = context.overflowType match {
-      case Grow => f -& g
-      case Wrap => f -% g
+      case Grow => throw DspException("OverflowType Grow is not supported for UInt subtraction")
+      case Wrap => f.asSInt -% g.asSInt
```

Comment:
@shunshou I guess I will got with the ```f -% g``` as it seems more direct and seems to work in all my tests